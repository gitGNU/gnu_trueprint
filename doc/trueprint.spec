Summary: Prints the source files in PostScript
Name: trueprint
Version: 5.3
Release: 1
Source: ftp://ftp.gnu.org/pub/gnu/trueprint/trueprint-%{PACKAGE_VERSION}.tar.gz
Copyright: GPL
Group: Applications/Text
Vendor: Jan "Yenya" Kasprzak <kas@fi.muni.cz>
Packager: Jan "Yenya" Kasprzak <kas@fi.muni.cz>
BuildRoot: /var/tmp/trueprint-root
Prereq: info
Obsoletes: Trueprint
%description

Trueprint is a general purpose program printing program.  It tries to produce
everything that anybody could need in a program printout without the need for
large numbers of switches or pipelines.

Trueprint can currently handle C, C++, Verilog, shell (including ksh),
perl, pascal, listing files and plain text files.

%changelog
* Tue Mar 20 2001 Lezz Giles <gilesfamily@mediaone.net>
- Updated to the 5.3 release.

* Wed Feb 16 2000 Lezz Giles <gilesfamily@mediaone.net>
- Updated to the 5.2.1 release.

* Wed Nov 3 1999 Lezz Giles <gilesfamily@mediaone.net>
- Updated to the 5.2 release.

* Sat Oct 30 1999 Jan "Yenya" Kasprzak <kas@fi.muni.cz>
- Updated to the 5.1 release.
- Strip trueprint (-s added to CFLAGS).

* Tue Jul 27 1999 Jan "Yenya" Kasprzak <kas@fi.muni.cz>
- Initial release

%prep
%setup -n trueprint-%{PACKAGE_VERSION}

%build

CFLAGS="$RPM_OPT_FLAGS" ./configure --prefix=/usr
make

%install
rm -rf ${RPM_BUILD_ROOT}
mkdir -p ${RPM_BUILD_ROOT}/usr/{bin,lib,info,man/man1}
make prefix=${RPM_BUILD_ROOT}/usr install
strip ${RPM_BUILD_ROOT}/usr/bin/trueprint

%post
if [ $1 = 1 ]
then
	/sbin/install-info --dir-file=/usr/info/dir /usr/info/trueprint.info
fi

%postun
if [ $1 = 0 ]
then
	/sbin/install-info --dir-file=/usr/info/dir --remove /usr/info/trueprint.info
fi

%files
%attr(0755,root,root) /usr/bin/trueprint
%attr(0644,root,root) %doc /usr/man/man1/trueprint.1
%attr(0644,root,root) %config(noreplace) /usr/lib/printers
%attr(0644,root,root) %doc /usr/info/trueprint.info
%attr(0644,root,root) %doc BUGS NEWS README
