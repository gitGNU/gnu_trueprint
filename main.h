/*
 * Include file:
 *	main.h
 */

/*
 * Globally available externs
 */
extern unsigned int	no_of_files;
extern char             *current_filename;
extern short		pass;
extern unsigned int	file_number;
extern int		page_length;
extern int		page_width;
extern long		total_pages;
extern boolean		use_environment;
