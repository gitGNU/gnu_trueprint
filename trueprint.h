/*
 * Include file:
 *	trueprint.h
 */

#define CMD_NAME "trueprint"

/*
 * TP_ADMIN_USER contains the mail address of the administrator.
 * You should change this address!
 */
#define TP_ADMIN_USER "root@localhost"

/*
 * Default settings for options, only needed if the defaults that are
 * hard-coded are not good enough.
 */
#define DEFAULT_OPTS ""

/*
 * LOGTOOL is set to the tool to be used to log invocations of trueprint.  To turn
 * off logging, comment this line out
 */
/* #define LOGTOOL ... */

/*
 * Maximum length of a command to be passed to system()
 */
#define COMMAND_LEN	2000

/*
 * Maximum length of an input or output line for diff, indexes, etc.
 */
#define INPUT_LINE_LEN  200

/*
 * Maximum length of a symbol, in the language parsing stuff
 */
#define SYMBOL_LEN      70

/*
 * Maximum line length of an output line for source code
 */
#define MAXLINELENGTH	2000

/*
 * Buffer size - input is read in in blocks this size
 */
#define BUFFER_SIZE 4096

/*
 * Postscript max and min values
 */
#define MAXPOINTSIZE 50
#define MINPOINTSIZE 5
#define MAXINTERLINE_GAP 10
#define MININTERLINE_GAP 0

/*
 * Maximum size of greenlines
 */
#define MAXGREENLINES 10

/*
 * HOLEPUNCH_WIDTH is the space, in points, left blank for holepunches.
 * 54 points is 3/4"
 */
#define HOLEPUNCH_WIDTH 54

/*
 * HOLEPUNCH_HEIGHT is the space, in points, left blank for two-hole 
 * holepunches (that punch at the top of the page).
 * 36 points is 1/2"
 */
#define HOLEPUNCH_HEIGHT 36

/*
 * Characters that we can break lines on.  BREAKSLENGTH is set to the
 * number of these characters
 */
#define BREAKS	";({,. 	"
#define BREAKSLENGTH	7

#define PUTS(x)	(void)fputs(x,stdout)

#define BACKSPACE (char)8

typedef enum {
	CHAR_NORMAL,
	CHAR_ITALIC,
        CHAR_BOLD,
	CHAR_UNDERLINE
} char_status;

typedef enum {
	STREAM_EMPTY_FILE = -1,
	STREAM_OK = 0x0,
	STREAM_PAGE_END = 0x1,
	STREAM_FUNCTION_END = 0x2,
	STREAM_FILE_END = 0x4
} stream_status;

#undef TRUE
#undef FALSE

typedef enum {
	UNSET = -1,
	FALSE = 0,
	TRUE = 1
} boolean;
