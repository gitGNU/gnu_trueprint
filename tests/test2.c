/* @(#)getopt.c */

#define _POSIX_SOURCE

#include <stdio.h>
#include <string.h>

#include "trueprint.h"
#include "main.h"

/*
 * get option letter from argument vector
 */
int		optind = 1;		/* index into parent argv vector */
char		*optarg;		/* argument associated with option */

static int	optopt;			/* character checked for validity */

int
getopt(int nargc, char **nargv, char *ostr)
{
	register char	*oli;		/* option letter list index */
	static char	*place = "";	/* option letter processing */

	if(!*place) {			/* update scanning pointer */
		if(optind >= nargc || *(place = nargv[optind]) != '-' || !*++place) {
		  place = "";
		  return(EOF);
		}
		if (*place == '-') {	/* found "--" */
		  ++optind;
		  place = "";
		  return EOF;
		}
	}				/* option letter okay? */
	if ((optopt = (int)*place++) == (int)':' || !(oli = strchr(ostr,optopt))) {
		if(!*place) ++optind;
		(void)fprintf(stderr, "%s: illegal option -- %c\n", cmd_name, optopt);
		return '?';
	}
	if (*++oli != ':') {		/* don't need argument */
		optarg = NULL;
		if (!*place) ++optind;
	} else {				/* need an argument */
		if (*place) {			/* no white space */
			optarg = place;
		} else if (nargc <= ++optind) {	/* no arg */
			place = "";
			(void)fprintf(stderr, "%s: option requires an argument -- %c\n", cmd_name, optopt);
			optopt = '?';
		} else {
			optarg = nargv[optind];	/* white space */
		}
		place = "";
		++optind;
	}
	return optopt;			/* dump back option letter */
}
