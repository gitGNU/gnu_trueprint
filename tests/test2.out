%!PS-Adobe-2.0
statusdict /setduplexmode known {
  false statusdict /setduplexmode get exec
} if
/Ps	10 def
/SPs	7 def
/Lh	11 def
/Rm	575 def
/Tm	746 def
/Bh	15 def
/Bf	15 def
/CF	/Courier findfont Ps scalefont def
/IF	/Courier-Oblique findfont Ps scalefont def
/BF	/Courier-Bold findfont Ps scalefont def
/CFs	/Courier findfont SPs scalefont def
/IFs	/Courier-Oblique findfont SPs scalefont def
/BFs	/Courier-Bold findfont SPs scalefont def
/HF	/Helvetica findfont Ps scalefont def
/Li CFs setfont (1234567890) stringwidth pop def
/Nl	{ /Vpos Vpos Lh sub def } def
/Lpt	{ 0 Vpos moveto } def
/Gb	{
		} def
/Ip	{ Gb .5 setlinewidth
		0 Tm moveto 0 Bh neg rlineto Rm 0 rlineto 0 Bh rlineto closepath
		gsave .98 setgray fill grestore stroke
		0 0 moveto 0 Bf rlineto Rm 0 rlineto 0 Bf neg rlineto closepath
		gsave .98 setgray fill grestore stroke
		0 Bf moveto 0 Tm Bh sub lineto stroke newpath
		} def
/Cp	{ Ip .3 setlinewidth newpath
		Li 0 Bf add moveto Li Tm Bh sub lineto stroke newpath
		} def
/So	{ gsave dup stringwidth pop Ps 3 div 0 exch rmoveto 0 rlineto fill grestore } def
/Ul	{ gsave	dup stringwidth pop 0 -1 rmoveto 0 rlineto fill grestore } def
/Bs	{ gsave	dup show grestore 0.5 0.5 rmoveto show } def
<</PageSize [605 806]>> setpagedevice
%%Page: Cover 1
70 70 moveto
/Helvetica findfont 10 scalefont setfont
(Trueprint 3.6.5) show
70 725 moveto
/Helvetica-Bold findfont 20 scalefont setfont
(For: testuser) show
70 700 moveto
(Printed on: Fri Apr 16 03:00:00 1982) show
70 675 moveto
(Last page number: 3) show
70 650 moveto (File: test1.c) show
70 625 moveto (File: test2.c) show
showpage
%%Page: 1 2
15 30 translate
Ip HF setfont /Ypos Tm Bh sub Ps 2 div add def
Ps Ypos moveto
(Fri Apr 16 03:00:00 1982) show currentpoint pop dup Rm Ps sub (Page 1) stringwidth pop sub dup Ypos moveto
(Page 1) show
exch sub 2 div add (Function Index) stringwidth pop 2 div sub Ypos moveto
(Function Index) show
HF setfont /Ypos Ps 2 div def
Ps Ypos moveto
(Fri Apr 16 03:00:00 1982) show currentpoint pop dup Rm Ps sub (Page 1) stringwidth pop sub dup Ypos moveto
(Page 1) show
exch sub 2 div add (Function Index) stringwidth pop 2 div sub Ypos moveto
(Function Index) show
CF setfont /Vpos Tm Bh sub Ps sub def
Lpt(          closepipe                   2  (test1.c)
) show Nl
Lpt(          fclosepipe..................2  (test1.c)) show Nl
Lpt(          fopenpipe                   2  (test1.c)
) show Nl
Lpt(          getopt......................3  (test2.c)) show Nl
Lpt(          openpipe                    1  (test1.c)
) show Nl
showpage
%%Page: 2 3
15 30 translate
Ip HF setfont /Ypos Tm Bh sub Ps 2 div add def
Ps Ypos moveto
(Fri Apr 16 03:00:00 1982) show currentpoint pop dup Rm Ps sub (Page 1) stringwidth pop sub dup Ypos moveto
(Page 1) show
exch sub 2 div add (File Index) stringwidth pop 2 div sub Ypos moveto
(File Index) show
HF setfont /Ypos Ps 2 div def
Ps Ypos moveto
(Fri Apr 16 03:00:00 1982) show currentpoint pop dup Rm Ps sub (Page 1) stringwidth pop sub dup Ypos moveto
(Page 1) show
exch sub 2 div add (File Index) stringwidth pop 2 div sub Ypos moveto
(File Index) show
CF setfont /Vpos Tm Bh sub Ps sub def
          Lpt(    test1.c                      1) BF setfont show CF setfont Nl
          Lpt(          closepipe...................2) show Nl
          Lpt(          fclosepipe                  2) show Nl
          Lpt(          fopenpipe...................2) show Nl
          Lpt(          openpipe                    1) show Nl
          Lpt(    test2.c                      3) BF setfont show CF setfont Nl
          Lpt(          getopt                      3) show Nl
          showpage
%%Page: 3 4
15 30 translate
Cp HF setfont /Ypos Tm Bh sub Ps 2 div add def
Ps Ypos moveto
(Fri Apr 16 03:00:00 1982) show currentpoint pop dup Rm Ps sub (Page 1 of 3) stringwidth pop sub dup Ypos moveto
(Page 1 of 3) show
exch sub 2 div add (openpipe) stringwidth pop 2 div sub Ypos moveto
(openpipe) show
HF setfont /Ypos Ps 2 div def
Ps Ypos moveto
(Fri Apr 16 03:00:00 1982) show currentpoint pop dup Rm Ps sub (Page 1 of 3) stringwidth pop sub dup Ypos moveto
(Page 1 of 3) show
exch sub 2 div add (test1.c 1) stringwidth pop 2 div sub Ypos moveto
(test1.c 1) show
/Vpos Tm Bh sub Ps sub def
Lpt CFs setfont (     1     ) show (/*) IF setfont show  Nl
Lpt CFs setfont (     2     ) show ( * Source file:) IF setfont show  Nl
Lpt CFs setfont (     3     ) show ( *      openpipe.c) IF setfont show  Nl
Lpt CFs setfont (     4     ) show ( *) IF setfont show  Nl
Lpt CFs setfont (     5     ) show ( * Contains openpipe - a replacement for the non-POSIX function popen.) IF setfont show  Nl
Lpt CFs setfont (     6     ) show ( * This returns an int rather than a FILE * to make it general purpose -) IF setfont show  Nl
Lpt CFs setfont (     7     ) show ( * fdopen can be used outside to convert this to a FILE *) IF setfont show  Nl
Lpt CFs setfont (     8     ) show ( */) IF setfont show  Nl
Lpt CFs setfont (           ) show () IF setfont show  Nl
Lpt CFs setfont (    10     ) show (#define _POSIX_SOURCE) CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    12     ) show (#include <errno.h>) CF setfont show  Nl
Lpt CFs setfont (    13     ) show (#include <stdio.h>) CF setfont show  Nl
Lpt CFs setfont (    14     ) show (#include <stdlib.h>) CF setfont show  Nl
Lpt CFs setfont (    15     ) show (#include <string.h>) CF setfont show  Nl
Lpt CFs setfont (    16     ) show (#include <sys/types.h>) CF setfont show  Nl
Lpt CFs setfont (    17     ) show (#include <sys/wait.h>) CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    19     ) show (#ifdef _ultrix) CF setfont show  Nl
Lpt CFs setfont (    20     ) show (extern int close\(int fildes\);) CF setfont show  Nl
Lpt CFs setfont (    21     ) show (extern int dup\(int fildes\);) CF setfont show  Nl
Lpt CFs setfont (    22     ) show (extern pid_t fork\(void\);) CF setfont show  Nl
Lpt CFs setfont (    23     ) show (extern int pipe\(int fildes[2]\);) CF setfont show  Nl
Lpt CFs setfont (    24     ) show (#else) CF setfont show  Nl
Lpt CFs setfont (    25     ) show (#include <unistd.h>) CF setfont show  Nl
Lpt CFs setfont (    26     ) show (#endif) CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    28     ) show (#include "trueprint.h") CF setfont show  Nl
Lpt CFs setfont (    29     ) show (#include "main.h") CF setfont show  Nl
Lpt CFs setfont (    30     ) show (#include "openpipe.h") CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    32     ) show (int) CF setfont show  Nl
Lpt CFs setfont (    33     ) show (openpipe) BF setfont show (\(const char *command, char *mode\)) CF setfont show  Nl
Lpt CFs setfont (           ) show () BF setfont show  Nl
Lpt CFs setfont (    35  1  ) show ({) CF setfont show  Nl
Lpt CFs setfont (    36  1  ) show (  int fildes[2];) CF setfont show  Nl
Lpt CFs setfont (    37  1  ) show (  int parent_pipe_end;) CF setfont show  Nl
Lpt CFs setfont (    38  1  ) show (  int child_pipe_end;) CF setfont show  Nl
Lpt CFs setfont (    39  1  ) show (  char *child_pipe_mode;) CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    41  2  ) show (  if \(strcmp\(mode, "w"\) == 0\) {) CF setfont show  Nl
Lpt CFs setfont (    42  2  ) show (    parent_pipe_end = 1; child_pipe_end = 0;) CF setfont show  Nl
Lpt CFs setfont (    43  2  ) show (    child_pipe_mode = "read";) CF setfont show  Nl
Lpt CFs setfont (    44  2  ) show (  } else if \(strcmp\(mode, "r"\) == 0\) {) CF setfont show  Nl
Lpt CFs setfont (    45  2  ) show (    parent_pipe_end = 0; child_pipe_end = 1;) CF setfont show  Nl
Lpt CFs setfont (    46  2  ) show (    child_pipe_mode = "write";) CF setfont show  Nl
Lpt CFs setfont (    47  2  ) show (  } else {) CF setfont show  Nl
Lpt CFs setfont (    48  2  ) show (    \(void\)fprintf\(stderr, "%s: Internal error: popen\() CF setfont show  Nl
Lpt CFs setfont (           ) show (                                          \) called with bad mode %s", cmd_name, mode\);) CF setfont show  Nl
Lpt CFs setfont (    49  2  ) show (    exit\(2\);) CF setfont show  Nl
Lpt CFs setfont (    50  1  ) show (  }) CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    52  2  ) show (  if \(pipe\(fildes\) != 0\) {) CF setfont show  Nl
Lpt CFs setfont (    53  2  ) show (    \(void\)fprintf\(stderr, "%s: Cannot create pipe, %s\\n", cmd_name, strerror\(errno\)\);) CF setfont show  Nl
Lpt CFs setfont (    54  2  ) show (    exit\(2\);) CF setfont show  Nl
Lpt CFs setfont (    55  1  ) show (  }) CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    57  2  ) show (  switch \(fork\(\)\) {) CF setfont show  Nl
Lpt CFs setfont (    58  2  ) show (  case -1:) CF setfont show  Nl
Lpt CFs setfont (    59  2  ) show (    ) CF setfont show (/* Error */) IF setfont show  Nl
Lpt CFs setfont (    60  2  ) show (    \(void\)fprintf\(stderr, "%s: Cannot fork, %s\\n", cmd_name, strerror\(errno\)\);) CF setfont show  Nl
Lpt CFs setfont (    61  2  ) show (    exit\(2\);) CF setfont show  Nl
Lpt CFs setfont (    62  2  ) show (    ) CF setfont show (/*NOTREACHED*/) IF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    64  2  ) show (  case 0:) CF setfont show  Nl
showpage
%%Page: 4 5
15 30 translate
Cp HF setfont /Ypos Tm Bh sub Ps 2 div add def
Ps Ypos moveto
(Fri Apr 16 03:00:00 1982) show currentpoint pop dup Rm Ps sub (Page 2 of 3) stringwidth pop sub dup Ypos moveto
(Page 2 of 3) show
exch sub 2 div add (openpipe) stringwidth pop 2 div sub Ypos moveto
(openpipe) show
HF setfont /Ypos Ps 2 div def
Ps Ypos moveto
(Fri Apr 16 03:00:00 1982) show currentpoint pop dup Rm Ps sub (Page 2 of 3) stringwidth pop sub dup Ypos moveto
(Page 2 of 3) show
exch sub 2 div add (test1.c 2) stringwidth pop 2 div sub Ypos moveto
(test1.c 2) show
/Vpos Tm Bh sub Ps sub def
Lpt CFs setfont (    65  2  ) show (    ) CF setfont show (/* Child process */) IF setfont show  Nl
Lpt CFs setfont (    66  2  ) show (    if \(!\(\(close\(fildes[parent_pipe_end]\) == 0\) &&) CF setfont show  Nl
Lpt CFs setfont (    67  2  ) show (         \(close\(child_pipe_end\) == 0\) &&) CF setfont show  Nl
Lpt CFs setfont (    68  3  ) show (         \(dup\(fildes[child_pipe_end]\) == child_pipe_end\)\)\) {) CF setfont show  Nl
Lpt CFs setfont (    69  3  ) show (      \(void\)fprintf\() CF setfont show  Nl
Lpt CFs setfont (           ) show ( stderr,"%s: Cannot redirect %s for child, %s\\n", cmd_name, child_pipe_mode, strerror\() CF setfont show  Nl
Lpt CFs setfont (           ) show (                                                                              errno\)\);) CF setfont show  Nl
Lpt CFs setfont (    70  3  ) show (      exit\(2\);) CF setfont show  Nl
Lpt CFs setfont (    71  2  ) show (    }) CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    73  3  ) show (    if \(system\(command\) < 0\) {) CF setfont show  Nl
Lpt CFs setfont (    74  3  ) show (      \(void\)fprintf\(stderr,"%s: Cannot start pipe, %s\\n",cmd_name, strerror\(errno\)\);) CF setfont show  Nl
Lpt CFs setfont (    75  3  ) show (      exit\(2\);) CF setfont show  Nl
Lpt CFs setfont (    76  2  ) show (    } else exit\(0\);) CF setfont show  Nl
Lpt CFs setfont (    77  2  ) show (    ) CF setfont show (/*NOTREACHED*/) IF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    79  2  ) show (  default:) CF setfont show  Nl
Lpt CFs setfont (    80  2  ) show (    ) CF setfont show (/* Parent process */) IF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    82  3  ) show (    if \(close\(fildes[child_pipe_end]\) != 0\) {) CF setfont show  Nl
Lpt CFs setfont (    83  3  ) show (      \(void\)fprintf\() CF setfont show  Nl
Lpt CFs setfont (           ) show (  stderr,"%s: Cannot close %s end of pipe for parent, %s\\n",cmd_name, child_pipe_mode,) CF setfont show  Nl
Lpt CFs setfont (           ) show (                                                                     strerror\(errno\)\);) CF setfont show  Nl
Lpt CFs setfont (    84  3  ) show (      exit\(2\);) CF setfont show  Nl
Lpt CFs setfont (    85  2  ) show (    }) CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    87  2  ) show (    return fildes[parent_pipe_end];) CF setfont show  Nl
Lpt CFs setfont (    88  1  ) show (  }) CF setfont show  Nl
Lpt CFs setfont (    89     ) show (}) CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    91     ) show (FILE *) CF setfont show  Nl
Lpt CFs setfont (    92     ) show (fopenpipe) BF setfont show (\(const char *command, char *mode\)) CF setfont show  Nl
Lpt CFs setfont (           ) show () BF setfont show  Nl
Lpt CFs setfont (    94  1  ) show ({) CF setfont show  Nl
Lpt CFs setfont (    95  1  ) show (  int handle;) CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    97  1  ) show (  handle = openpipe\(command, mode\);) CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    99  1  ) show (  return fdopen\(handle, mode\);) CF setfont show  Nl
Lpt CFs setfont (   100     ) show (}) CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (   102     ) show (void) CF setfont show  Nl
Lpt CFs setfont (   103     ) show (closepipe) BF setfont show (\(int handle\)) CF setfont show  Nl
Lpt CFs setfont (           ) show () BF setfont show  Nl
Lpt CFs setfont (   105  1  ) show ({) CF setfont show  Nl
Lpt CFs setfont (   106  1  ) show (  int statloc;) CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (   108  1  ) show (  \(void\)close\(handle\);) CF setfont show  Nl
Lpt CFs setfont (   109  1  ) show (  \(void\)waitpid\(-1, &statloc, WNOHANG\);) CF setfont show  Nl
Lpt CFs setfont (   110     ) show (}) CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (   112     ) show (void) CF setfont show  Nl
Lpt CFs setfont (   113     ) show (fclosepipe) BF setfont show (\(FILE *fp\)) CF setfont show  Nl
Lpt CFs setfont (           ) show () BF setfont show  Nl
Lpt CFs setfont (   115  1  ) show ({) CF setfont show  Nl
Lpt CFs setfont (   116  1  ) show (  closepipe\(fileno\(fp\)\);) CF setfont show  Nl
Lpt CFs setfont (   117     ) show (}) CF setfont show  Nl
showpage
%%Page: 5 6
15 30 translate
Cp HF setfont /Ypos Tm Bh sub Ps 2 div add def
Ps Ypos moveto
(Fri Apr 16 03:00:00 1982) show currentpoint pop dup Rm Ps sub (Page 3 of 3) stringwidth pop sub dup Ypos moveto
(Page 3 of 3) show
exch sub 2 div add (getopt) stringwidth pop 2 div sub Ypos moveto
(getopt) show
HF setfont /Ypos Ps 2 div def
Ps Ypos moveto
(Fri Apr 16 03:00:00 1982) show currentpoint pop dup Rm Ps sub (Page 3 of 3) stringwidth pop sub dup Ypos moveto
(Page 3 of 3) show
exch sub 2 div add (test2.c 1) stringwidth pop 2 div sub Ypos moveto
(test2.c 1) show
/Vpos Tm Bh sub Ps sub def
Lpt CFs setfont (     1     ) show (/* @\(#\)getopt.c */) IF setfont show  Nl
Lpt CFs setfont (           ) show () IF setfont show  Nl
Lpt CFs setfont (     3     ) show (#define _POSIX_SOURCE) CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (     5     ) show (#include <stdio.h>) CF setfont show  Nl
Lpt CFs setfont (     6     ) show (#include <string.h>) CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (     8     ) show (#include "trueprint.h") CF setfont show  Nl
Lpt CFs setfont (     9     ) show (#include "main.h") CF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    11     ) show (/*) IF setfont show  Nl
Lpt CFs setfont (    12     ) show ( * get option letter from argument vector) IF setfont show  Nl
Lpt CFs setfont (    13     ) show ( */) IF setfont show  Nl
Lpt CFs setfont (    14     ) show (int             optind = 1;             ) CF setfont show (/* index into parent argv vector */) IF setfont show  Nl
Lpt CFs setfont (    15     ) show (char            *optarg;                ) CF setfont show (/* argument associated with option */) IF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    17     ) show (static int      optopt;                 ) CF setfont show (/* character checked for validity */) IF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    19     ) show (int) CF setfont show  Nl
Lpt CFs setfont (    20     ) show (getopt) BF setfont show (\(int nargc, char **nargv, char *ostr\)) CF setfont show  Nl
Lpt CFs setfont (    21  1  ) show ({) CF setfont show  Nl
Lpt CFs setfont (    22  1  ) show (        register char   *oli;           ) CF setfont show (/* option letter list index */) IF setfont show  Nl
Lpt CFs setfont (    23  1  ) show (        static char     *place = "";    ) CF setfont show (/* option letter processing */) IF setfont show  Nl
Lpt CFs setfont (           ) show () CF setfont show  Nl
Lpt CFs setfont (    25  2  ) show (        if\(!*place\) {                   ) CF setfont show (/* update scanning pointer */) IF setfont show  Nl
Lpt CFs setfont (    26  3  ) show (                if\(optind >= nargc || *\(place = nargv[optind]\) != '-' || !*++place\) {) CF setfont show  Nl
Lpt CFs setfont (    27  3  ) show (                  place = "";) CF setfont show  Nl
Lpt CFs setfont (    28  3  ) show (                  return\(EOF\);) CF setfont show  Nl
Lpt CFs setfont (    29  2  ) show (                }) CF setfont show  Nl
Lpt CFs setfont (    30  3  ) show (                if \(*place == '-'\) {    ) CF setfont show (/* found "--" */) IF setfont show  Nl
Lpt CFs setfont (    31  3  ) show (                  ++optind;) CF setfont show  Nl
Lpt CFs setfont (    32  3  ) show (                  place = "";) CF setfont show  Nl
Lpt CFs setfont (    33  3  ) show (                  return EOF;) CF setfont show  Nl
Lpt CFs setfont (    34  2  ) show (                }) CF setfont show  Nl
Lpt CFs setfont (    35  1  ) show (        }                               ) CF setfont show (/* option letter okay? */) IF setfont show  Nl
Lpt CFs setfont (    36  2  ) show (        if \(\(optopt = \(int\)*place++\) == \(int\)':' || !\(oli = strchr\(ostr,optopt\)\)\) {) CF setfont show  Nl
Lpt CFs setfont (    37  2  ) show (                if\(!*place\) ++optind;) CF setfont show  Nl
Lpt CFs setfont (    38  2  ) show (                \(void\)fprintf\(stderr, "%s: illegal option -- %c\\n", cmd_name, optopt\);) CF setfont show  Nl
Lpt CFs setfont (    39  2  ) show (                return '?';) CF setfont show  Nl
Lpt CFs setfont (    40  1  ) show (        }) CF setfont show  Nl
Lpt CFs setfont (    41  2  ) show (        if \(*++oli != ':'\) {            ) CF setfont show (/* don't need argument */) IF setfont show  Nl
Lpt CFs setfont (    42  2  ) show (                optarg = NULL;) CF setfont show  Nl
Lpt CFs setfont (    43  2  ) show (                if \(!*place\) ++optind;) CF setfont show  Nl
Lpt CFs setfont (    44  2  ) show (        } else {                                ) CF setfont show (/* need an argument */) IF setfont show  Nl
Lpt CFs setfont (    45  3  ) show (                if \(*place\) {                   ) CF setfont show (/* no white space */) IF setfont show  Nl
Lpt CFs setfont (    46  3  ) show (                        optarg = place;) CF setfont show  Nl
Lpt CFs setfont (    47  3  ) show (                } else if \(nargc <= ++optind\) { ) CF setfont show (/* no arg */) IF setfont show  Nl
Lpt CFs setfont (    48  3  ) show (                        place = "";) CF setfont show  Nl
Lpt CFs setfont (    49  3  ) show (                        \(void\)fprintf\() CF setfont show  Nl
Lpt CFs setfont (           ) show (                 stderr, "%s: option requires an argument -- %c\\n", cmd_name, optopt\);) CF setfont show  Nl
Lpt CFs setfont (    50  3  ) show (                        optopt = '?';) CF setfont show  Nl
Lpt CFs setfont (    51  3  ) show (                } else {) CF setfont show  Nl
Lpt CFs setfont (    52  3  ) show (                        optarg = nargv[optind]; ) CF setfont show (/* white space */) IF setfont show  Nl
Lpt CFs setfont (    53  2  ) show (                }) CF setfont show  Nl
Lpt CFs setfont (    54  2  ) show (                place = "";) CF setfont show  Nl
Lpt CFs setfont (    55  2  ) show (                ++optind;) CF setfont show  Nl
Lpt CFs setfont (    56  1  ) show (        }) CF setfont show  Nl
Lpt CFs setfont (    57  1  ) show (        return optopt;                  ) CF setfont show (/* dump back option letter */) IF setfont show  Nl
Lpt CFs setfont (    58     ) show (}) CF setfont show  Nl
showpage
%%Trailer
%%EOF
